src/Hosting/WindowsServices/src/WindowsServiceLifetime.cs

```diff
@@ -90,6 +91,12 @@ protected override void OnStop()
             _delayStop.Wait(_hostOptions.ShutdownTimeout);
             base.OnStop();
         }
+        
+        protected override void OnShutdown()
+        {
+             ApplicationLifetime.StopApplication();
```

Comment:
If OnShutdown blocks will that delay shutdown? Does the application have time to actually shut down before the OS kills it? Compare to OnStop.

Reply:
Good point. will fix

Reply:
good point, will fix it.